python/ql/test/query-tests/Security/CWE-022/tarslip.py

```diff
@@ -50,3 +50,19 @@ def safemembers(members):
 
 tar = tarfile.open(unsafe_filename_tar)
 tar.extractall(members=safemembers(tar))
+
+
+# Wrong sanitizer (is missing not)
+tar = tarfile.open(unsafe_filename_tar)
+for entry in tar:
+    if os.path.isabs(entry.name) or ".." in entry.name:
+        tar.extract(entry, "/tmp/unpack/")
+
+
+# OK Sanitized using not
+tar = tarfile.open(unsafe_filename_tar)
+for entry in tar:
+    # using `if not (os.path.isabs(entry.name) or ".." in entry.name):`
+    # would make the sanitizer work, but for the wrong reasons since out library is a bit broken.
```

Comment:
Typo: `out`
Also maybe add a separate test case for this so we can keep track of when it is fixed?

Reply:
Right, so the problem is that using `if not (os.path.isabs(entry.name) or False):` does by default not give the same result as using `if not (os.path.isabs(entry.name)`. Since this is fixed by handling `not` manually (as done in this PR), it's **really** hard to make a good test for it here. I tried to explain this in the code, since I was actually having trouble figuring out what old-me meant with this.